### PR TITLE
[MINOR][SQL] Combine the same codes in test cases

### PR DESCRIPTION
**Pull request by CarolinePeng**
## What changes were proposed in this pull request?

In the DDLSuit, there are four test cases have the same codes , writing a function can combine the same code.

## How was this patch tested?

existing tests.

### CODE REVIEW

**Comment by kiszk** [comment]
Good catch

**Comment by kiszk** [comment]
ok to test

File: core/src/main/scala/org/apache/spark/memory/ExecutionMemoryPool.scala [hunk 1]
```diff
@@ -37,7 +37,7 @@ import org.apache.spark.internal.Logging
  * tasks was performed by the ShuffleMemoryManager.
  *
  * @param lock a [[MemoryManager]] instance to synchronize on
- * @param memoryMode the type of memory tracked by this pool (on- or off-heap)
+ * @param memoryMode the type of memory tracked by this pool (on-heap or off-heap)
```
Comment (kiszk) [inline core/src/main/scala/org/apache/spark/memory/ExecutionMemoryPool.scala]: Is this change related to this PR?

File: core/src/main/scala/org/apache/spark/memory/StorageMemoryPool.scala [hunk 2]
```diff
@@ -28,7 +28,7 @@ import org.apache.spark.storage.memory.MemoryStore
  * (caching).
  *
  * @param lock a [[MemoryManager]] instance to synchronize on
- * @param memoryMode the type of memory tracked by this pool (on- or off-heap)
+ * @param memoryMode the type of memory tracked by this pool (on-heap or off-heap)
```
Comment (kiszk) [inline core/src/main/scala/org/apache/spark/memory/StorageMemoryPool.scala]: ditto

**Comment by heary-cao** [comment]
LGTM, cc @HyukjinKwon.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 3]
```diff
@@ -377,41 +377,42 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
     }
   }
 
-  test("CTAS a managed table with the existing empty directory") {
-    val tableLoc = new File(spark.sessionState.catalog.defaultTablePath(TableIdentifier("tab1")))
-    try {
+  protected def withTableLocation(tableNames: String)(f: File => Unit): Unit = {
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: How about `withEmptyDirInTablePath(dirName: String)`?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 4]
```diff
@@ -377,41 +377,42 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
...
+    val tableLoc =
+      new File(spark.sessionState.catalog.defaultTablePath(TableIdentifier(tableNames)))
+    try
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: nit `try {`

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 5]
```diff
@@ -439,31 +440,22 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
...
         }.getMessage
         val expectedMsg = "Can not rename the managed table('`tab1`'). The associated location"
         assert(ex.contains(expectedMsg))
       }
-    } finally {
-      waitForTasksToFinish()
-      Utils.deleteRecursively(tableLoc)
     }
   }
-
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: nit: revert this

**Review by srowen — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala (hunk 3 above)
Comment (CarolinePeng) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: Ok,thank you !

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 6]
```diff
@@ -377,41 +377,41 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
     }
   }
 
-  test("CTAS a managed table with the existing empty directory") {
-    val tableLoc = new File(spark.sessionState.catalog.defaultTablePath(TableIdentifier("tab1")))
+  protected def withEmptyDirInTablePath(dirName: String)(f: File => Unit): Unit = {
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: private?

**Comment by maropu** [comment]
LGTM except for minor comments

**Comment by maropu** [comment]
thanks, merging to master!
